circuits/src/zk_circuits/valid_match_mpc.rs

```diff
@@ -536,6 +539,21 @@ impl<'a, N: 'a + MpcNetwork + Send, S: SharedValueSource<Scalar>> MultiProverCir
                     as CircuitBaseType>::VarType<Variable>,
         cs: &mut CS,
     ) -> Result<(), R1CSError> {
-        Self::matching_engine_check_single_prover(witness, cs)
+        ValidMatchMpcSingleProver::matching_engine_check_single_prover(witness, cs)
```

Comment:
ok giga nit but can we do:
```suggestion
        ValidMatchMpcSingleProver::apply_constraints(witness, statement, cs)
```
just feels like a proper use of the abstraction...

feel free to tell me to stfu

Reply:
yeah that's clean